src/gmail.js

```diff
@@ -3291,6 +3306,40 @@ var Gmail = function(localJQuery) {
         return undefined;
     };
 
+    /**
+     * Returns data about the currently visible messages available in the DOM:
+     * {
+     *    from: {
+     *      name: string,
+     *      email: string,
+     *    },
+     *    summary: string, // subject and email summary
+     *    thread_id: string,
+     *    $el: HTMLElement,
+     * }
+     */
+    api.dom.visible_messages = function() {
+        const ret = [];
+        $('tbody>tr.zA[draggable="true"]:visible', api.dom.inbox_content())
+            .each((index, msgEle) => {
+                const nameAndEmail = $('*[email][name]', msgEle);
+                const linkAndSubject = $('*[role=link]', msgEle);
+                // example value: #thread-f:1638756560099919527|msg-f:1638756560099919527"
+                const idNode = msgEle.querySelector("span[data-thread-id]");
+                ret.push({
+                    from: {
+                        name: nameAndEmail.attr('name'),
+                        email: nameAndEmail.attr('email'),
+                    },
+                    summary: linkAndSubject[0].innerText,
+                    thread_id: api.helper.clean_thread_id(idNode.dataset.threadId || ""),
+                    legacyMessageId:  idNode.dataset.legacyMessageId,
```

Comment:
We've used the identifier `legacy_email_id` other places in the code, so for consistency it would be nice to use here as well.

Reply:
I see you've probably chosen this identifier due to me being imprecise in my previous comment, and I apologize for that.

But really, if we just make it `legacy_email_id` everything else will be good :smile: